File: plus/6_draw_mixed2d.py
import gs
import gs.plus.render as render
import gs.plus.input as input

render.init(400, 300, "../pkg.core")

while not input.key_press(gs.InputDevice.KeyEscape):
	render.clear()
	render.text2d(20, 200, "UNDER", 64)
	render.triangle2d(0 + 75, 150 - 50, 100 + 75, 300 - 50, 200 + 75, 150 - 50, gs.Color.Red, gs.Color.Blue, gs.Color.Green)
	render.line2d(0, 0, 400, 300)
	render.triangle2d(200 - 75, 0 + 50, 300 - 75, 150 + 50, 400 - 75, 0 + 50, gs.Color.Red, gs.Color.Blue, gs.Color.Green)
	render.text2d(200, 50, "OVER", 64)
	render.set_blend_mode2d(render.BlendAdditive)
	render.text2d(120, 135, "ADDITIVE", 32, gs.Color.Blue)
	render.set_blend_mode2d(render.BlendOpaque)
	render.flip()

File: plus/6_draw_mixed2d.lua
hg = require("harfang")

hg.LoadPlugins()

plus = hg.GetPlus()
plus:RenderInit(400, 300)

while not plus:IsAppEnded() do
	plus:Clear()
	plus:Text2D(20, 200, "UNDER", 64)
	plus:Triangle2D(0 + 75, 150 - 50, 100 + 75, 300 - 50, 200 + 75, 150 - 50, hg.Color.Red, hg.Color.Blue, hg.Color.Green)
	plus:Line2D(0, 0, 400, 300)
	plus:Triangle2D(200 - 75, 0 + 50, 300 - 75, 150 + 50, 400 - 75, 0 + 50, hg.Color.Red, hg.Color.Blue, hg.Color.Green)
	plus:Text2D(200, 50, "OVER", 64)
	plus:SetBlend2D(hg.BlendAdditive)
	plus:Text2D(120, 135, "ADDITIVE", 32, hg.Color.Blue)
	plus:SetBlend2D(hg.BlendOpaque)
	plus:Flip()
	plus:EndFrame()
end

plus:RenderUninit()